include/tbl/cli_cmds.h

```diff
@@ -42,13 +42,24 @@
 
 /*lint -save -e525 -e539 */
 
+#ifdef CLI_LOG_FUNCS
+
+#  ifndef VSB_H_INCLUDED
+#    include "vsb.h"
+#  endif
+
+
+#undef CLI_LOG_FUNCS
+#endif
```

Comment:
This looks really out of place and while I had reserves regarding a logging callback, I can very likely be convinced that this is a good idea.

This `#ifdef` block however convinces me that this is not properly articulated.

While the sensitive flag belongs in `cli_cmd_desc`, the callback most likely belongs in `cli_proto` since there may be more than one implementation of the same command (essentially mcf and ccf).

Reply:
I initially thought of making the callback part of `cli_proto` as you suggest, but the thing is that MGT does not (and should not) have access to implementations of child cli commands. Thus, if MGT receives a command it does not know how to execute, it will be able to figure out that this command is sensitive by looping through the list of `cli_cmd_desc`s, but it won't know how to log it. Also I think that we should avoid having different implementations of the same command now that we have internal commands, we can have two separate commands and making the child one internal.

Reply:
see my last comment